### PR TITLE
[WIP] Torch 2.4 in docker images

### PR DESCRIPTION
**Pull request by snarayan21**
Adds torch 2.4 docker images.

Also reverts ci-testing to v0.1.2 so that dependencies are not automatically upgraded by uv.

Passing GPU tests (manual run): https://github.com/mosaicml/llm-foundry/actions/runs/10640011732

### CODE REVIEW

**Review by mvpatel2000 — APPROVED** [review]
@dakinggg to decide if we want to remove old ones, I think yes

**Comment by dakinggg** [comment]
@mvpatel2000 we do in 3 stages: (1) add the new images and bump the max pin, (2) add the new CI and make sure it passes, (3) bump the min pin and remove the older version

File: Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ ADD https://raw.githubusercontent.com/mosaicml/llm-foundry/$BRANCH_NAME/setup.py
 RUN rm setup.py
 
 # Install TransformerEngine
-RUN NVTE_FRAMEWORK=pytorch CMAKE_BUILD_PARALLEL_LEVEL=4 MAX_JOBS=4 pip install git+https://github.com/NVIDIA/TransformerEngine.git@b5a7c9f
+# RUN NVTE_FRAMEWORK=pytorch CMAKE_BUILD_PARALLEL_LEVEL=4 MAX_JOBS=4 pip install git+https://github.com/NVIDIA/TransformerEngine.git@b5a7c9f
```
Comment (irenedea) [inline Dockerfile:18]: delete completely?

Comment (snarayan21) [reply Dockerfile:18]: sry yeah am testing if te install works as a pip package
spoiler alert, it doesn't :/
marked this as a draft

**Review by dakinggg — APPROVED** [review]

**Pull request merged**